R/getShortTermChange.R

```diff
@@ -0,0 +1,146 @@
+        .check_assay_present(assay.type, x)
+        if(!.is_a_bool(rarefy)){
+            stop("'rarefy' must be TRUE or FALSE.", call. = FALSE)
+        }
+        if(!.is_a_bool(compositional)){
+            stop("'compositional' must be TRUE or FALSE.", call. = FALSE)
+        }
+        # Ensure that the provided depth is valid
+        if ( !is.null(depth) && depth > min(assay(x, assay.type), na.rm = TRUE) ) {
+            stop("Depth cannot be greater than the minimum number of counts in your data", call. = FALSE)
+        
+        }
+        ########################### Growth Metrics ############################
+        grwt <- .calculate_growth_metrics(x, assay.type = assay.type, 
+                                          rarefy = rarefy, 
+                                          compositional = compositional, 
+                                          depth = depth, ...)
+        # Clean and format growth matrics
+        grs.all <- .clean_growth_metrics(grwt, ...)
+        return(grs.all)
+    }
+)
+# wrapper to calculate growth matrix
+.calculate_growth_metrics <- function(x, assay.type, time.col = NULL, 
+                                      rarefy, compositional, depth, ...) {
+    ############################ Data Preparation ##############################
+    # Initialize the filtered object based on rarefy and compositional arguments
+    if (rarefy == TRUE && compositional == FALSE) {
+        message("rarefy is set to TRUE, calculating short term change using counts")
+        x <- rarefyAssay(x, assay.type = assay.type, depth = depth)
+        assay.type <- "subsampled"
+    } else if (rarefy == FALSE && compositional == FALSE) {
+        message("rarefy is set to FALSE, compositional==FALSE, using raw counts")
+        x <- x
+    } else if (rarefy == FALSE && compositional == TRUE) {
+        message("rarefy is set to FALSE, compositional==TRUE, using relative abundances")
+        x <- transformAssay(x, method = "relabundance", assay.type = assay.type)
+        assay.type <- "relabundance"
+    } else if (rarefy == TRUE && compositional == TRUE) {
+        stop("Both rarefy and compositional cannot be TRUE simultaneously", call. = FALSE)
+    }
+    # Reshape data and calcualte grwoth metrics
+    assay_data <- meltSE(x, assay.type = assay.type, add.col = time.col)
+    assay_data <- assay_data %>%
+        arrange( !!sym(time.col) ) %>%
+        group_by(FeatureID) %>%
+        mutate(
+            time_lag = !!sym(time.col) - lag( !!sym(time.col) ), 
+            growth_diff =!!sym(assay.type) - lag(!!sym(assay.type)),
+            growth_rate = (!!sym(assay.type) - lag(!!sym(assay.type))) / lag(!!sym(assay.type)),
+            var_abund = (!!sym(assay.type) - lag(!!sym(assay.type))) / time_lag
+        )
+    return(assay_data)
+}
+
+.clean_growth_metrics <- function(grwt, time.col = NULL, ...) {
+    # Calculate max growth
+    maxgrs <- grwt %>%
+        summarize(max.growth = max(growth_diff, na.rm = TRUE))
```

Comment:
The df has now column names with three different styles: "column_name" , "columname" and "column,name" --> use the first one

Reply:
or even col_name ?